src/snowflake/snowpark/types.py

```diff
@@ -41,12 +48,26 @@ def __repr__(self):
     def is_primitive(self):
         return True
 
+    def _fill_ast(self, ast: proto.SpDataType) -> None:
```

Comment:
I'm guessing that this would not raise module not found error since we are packaging `proto.SpDataType` with snowpark now. I mentioned this to @sfc-gh-lspiegelberg offline but we should add a merge-gate without proto and tzlocal installed to ensure when AST is disabled, tests work correctly. This can be added to daily precommit similar to https://github.com/snowflakedb/snowpark-python/actions/runs/11609891507/job/32328205853